proxy/src/coco/source.rs

```diff
@@ -197,6 +198,7 @@ pub struct TreeEntry {
 /// Will return [`error::Error`] if the project doesn't exist or a surf interaction fails.
 pub fn blob(
     browser: &mut Browser,
+    _peer_id: Option<&peer::PeerId>,
```

Comment:
I'm unsure if we need the `PeerId` at this scope, but that can be determined in another PR.

Reply:
@FintanH We want to scope by namespace (peer_id, "repository") whenever doing source queiries. Under what condition would we not need to be explicit about that scope when fetching a blob?

Reply:
Discussed on IRC. I didn't provide enough context here. Basically what I was trying to say that I'm not sure whether `blob` et al. should be aware of `PeerId` or if we make a `Browser` remote aware (where the `PeerId` is the remote). I need to evaluate whether this is useful for surf (my feeling is that it is) and then see what changes we can make. This is can be left as a reminder to do that work ;)